### PR TITLE
Add release notes for Barman 3.11.0

### PR DESCRIPTION
**Pull request by barthisrael**
References: BAR-248.

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -1,5 +1,185 @@
...
+    Postgres at the time the backup was taken. In JSON format, this is stored as
+    `server_information.data_checksums`.
+
+  - Add the “Backup method” field. This shows the backup method used for this
+    backup. In JSON format, this is stored as
+    `base_backup_information.backup_method`.
+
+  - Rename the field “Disk Usage” as “Backup Size”. The latter provides a more
+    comprehensive name which represents the size of the backup in the Barman
+    host. The JSON fields under `base_backup_information` were also renamed from
```
Comment (andremagui) [inline NEWS:58]:
```suggestion
    host. The JSON field under `base_backup_information` was also renamed from
```

File: AUTHORS [hunk 2]
```diff
@@ -1,6 +1,9 @@
 Barman maintainers (in alphabetical order):
 
+* Andre Marchesini de Aguiar
```
[the remark this answers is not in the document]
Comment (barthisrael) [reply AUTHORS]: We addressed this through #999

**Review by gcalacoci — APPROVED** [review]

File: NEWS [hunk 3]
```diff
@@ -1,5 +1,185 @@
...
+- Always set a timestamp with a time zone when configuring
+  `recovery_target_time` through `barman recover`. Previously, if no time zone
+  was explicitly set through `--target-time`, Barman would configure
+  `recovery_target_time` without a time zone in Postgres. Without a time zone,
+  Postgres would assume whatever is configured through `timezone` GUC in
+  Postgres. From now on Barman will issue a warning and configure
+  `recovery_target_time` with the time zone of the Barman host, if no time zone
+  is set by the user through `--target-time` option.
+
+- When recovering a backup with “no get wal” approach and `--target-lsn` is set,
```
Comment (martinmarques) [inline NEWS:118]:
```suggestion
- When recovering a backup with the “no get wal” approach and `--target-lsn` is set,
```

File: NEWS [hunk 4]
```diff
@@ -1,5 +1,185 @@
...
+  copy only the WAL files required to reach the configured target. Previously
+  Barman was copying all the WAL files from its archive to Postgres.
+
+- When recovering a backup with “no get wal” approach and `--target-immediate`
```
Comment (martinmarques) [inline NEWS:122]:
```suggestion
- When recovering a backup with the “no get wal” approach and `--target-immediate`
```

File: NEWS [hunk 5]
```diff
@@ -1,5 +1,185 @@
...
+  is set, copy only the WAL files required to reach the consistent point.
+  Previously Barman was copying all the WAL files from its archive to Postgres.
```
Comment (martinmarques) [inline NEWS:124]:
```suggestion
  Previously Barman would copy all the WAL files from its archive to Postgres.
```

File: NEWS [hunk 6]
```diff
@@ -1,5 +1,185 @@
...
+  was explicitly set through `--target-time`, Barman would configure
+  `recovery_target_time` without a time zone in Postgres. Without a time zone,
+  Postgres would assume whatever is configured through `timezone` GUC in
+  Postgres. From now on Barman will issue a warning and configure
+  `recovery_target_time` with the time zone of the Barman host, if no time zone
+  is set by the user through `--target-time` option.
+
+- When recovering a backup with “no get wal” approach and `--target-lsn` is set,
+  copy only the WAL files required to reach the configured target. Previously
+  Barman was copying all the WAL files from its archive to Postgres.
```
Comment (martinmarques) [inline NEWS:120]:
```suggestion
  Barman would copy all the WAL files from its archive to Postgres.
```

File: NEWS [hunk 7]
```diff
@@ -1,5 +1,185 @@
...
+
+- When recovering a backup with “no get wal” approach and `--target-immediate`
+  is set, copy only the WAL files required to reach the consistent point.
+  Previously Barman was copying all the WAL files from its archive to Postgres.
+
+- `barman-wal-restore` now moves WALs from the spool directory to `pg_wal`
+  instead of copying them. This can improve performance if the spool directory
+  and the `pg_wal` directory are in the same partition of the filesystem.
```
Comment (martinmarques) [inline NEWS:128]:
```suggestion
  and the `pg_wal` directory are in the same partition.
```

File: NEWS [hunk 8]
```diff
@@ -1,5 +1,185 @@
...
+
+- Add a keep-alive mechanism to rsync-based backups. Previously the Postgres
+  session created by Barman to run `pg_backup_start` and `pg_backup_stop` would
+  be idle for how long the base backup copy could take. That could lead to a
+  firewall or router dropping the connection because it was idle for a long
+  time. The keep-alive mechanism keeps sending heartbeat queries to Postgres
+  through that connection, thus reducing the likelihood of a connection
+  droppage. The interval between heartbeats can be controlled through the new
+  configuration option `keepalive_interval` and through the corresponding CLI
+  option `--keepalive-interval` of the `barman backup` command.
```
Comment (martinmarques) [inline NEWS:147]:
```suggestion
- Add a keep-alive mechanism to rsync-based backups. Previously the Postgres
  session created by Barman to run `pg_backup_start()` and `pg_backup_stop()` would
  stay idle for as long as the base backup copy would take. That could lead to a
  firewall or router dropping the connection because it was idle for a long
  time. The keep-alive mechanism sends heartbeat queries to Postgres
  through that connection, thus reducing the likelihood of a connection
  getting dropped. The interval between heartbeats can be controlled through the new
  configuration option `keepalive_interval` and the corresponding CLI
  option `--keepalive-interval` of the `barman backup` command.
```

File: NEWS [hunk 9]
```diff
@@ -1,5 +1,185 @@
...
+
+- Bug fixes:
+
+  - When recovering a backup with “no get wal” approach and `--target-time` is
```
Comment (martinmarques) [inline NEWS:151]:
```suggestion
  - When recovering a backup with the “no get wal” approach and `--target-time`
```

File: NEWS [hunk 10]
```diff
@@ -1,5 +1,185 @@
...
+    set, copy all WAL files. Previously Barman would attempt to “guess” the WAL
+    files required by Postgres to reach the configured target time. However,
+    the mechanism was not robust enough as it was based on the stats of the WAL
+    file in the Barman host (more specifically the creation time). For example:
+    if there were archiving or streaming lag between Postgres and Barman, that
+    could be enough for recovery to fail because Barman would miss to copy all
+    the required WAL files due to the weak check based on file stats.
+
+  - Pin `python-snappy` to `0.6.1` when running Barman through Python 3.6 or
+  older. Newer versions of `python-snappy` require `cramjam` on `2.7.0` or
```
Comment (martinmarques) [inline NEWS:161]:
```suggestion
  older. Newer versions of `python-snappy` require `cramjam` version `2.7.0` or
```

File: NEWS [hunk 11]
```diff
@@ -1,5 +1,185 @@
...
+
+    - Being unable to start `pg_receivewal` process because it is already
+      running.
+
+  - Fix and improve information about Python in `barman diagnose` output:
+
+    - The command now makes sure to use the correct Python interpreter where
+      Barman is installed when outputting the Python version through
+      `python_ver` JSON key. Previously, if a environment had multiple Python
+      installations and/or virtual environments, the output could be misleading.
```
Comment (martinmarques) [inline NEWS:178]: The first part is confusing. Which is the correct Python interpreter?

**Review by martinmarques — CHANGES_REQUESTED** [review]

File: NEWS (hunk 11 above)
Comment (barthisrael) [reply NEWS:178]: The same one that is is being used by Barman. Let me rephrase this a bit.

File: NEWS [hunk 12]
```diff
@@ -1,5 +1,185 @@
...
+
+  - When deleting a backup all the incremental backups depending on it, if any,
+    are also removed.
+
+  - Retention policies do not take incremental backups into consideration. As
+    incremental backups cannot be recovered without having the complete chain of
+    backups available up to the full backup, only full backups account for
+    retention policies.
+
+  - `barman recover` needs to combine the full with the chain of incremental
```
[the remark this answers is not in the document]
Comment (gcalacoci) [reply NEWS]: i think the whole period needs to be reworded @martinmarques @barthisrael

File: NEWS [hunk 13]
```diff
@@ -1,5 +1,185 @@
...
+    backups when recovering an incremental backup. The new CLI option
```
[the remark this answers is not in the document]
Comment (barthisrael) [reply NEWS:30]: Yeah, the problem is that currently we depend on ` LocalBackupInfo` in order to perform all the recovery steps.
So, we had to create a subclass of it called `SyntheticBackupInfo` which basically has the information of the incremental backup referenced by `barman recover command`, with a different path set.

File: NEWS [hunk 14]
```diff
@@ -1,5 +1,185 @@
...
+  - Add the “Estimated cluster size” field. This is useful to have an estimation
+    of the data directory size of a cluster when restoring a backup. It’s
+    particularly useful when recovering compressed backups or incremental
+    backups, situations where the backup size doesn’t reflect the size of the
+    data directory in Postgres. In JSON format, this is stored as
+    `cluster_size`.
+
+  - Add the “WAL summarizer” field. This field shows if `summarize_wal` was
+    enabled in Postgres at the time the backup was taken. In JSON format, this
+    is stored as `server_information.summarize_wal`.
```
[the remark this answers is not in the document]
Comment (barthisrael) [reply NEWS:46]: I think so

**Review by martinmarques — APPROVED** [review]
